lib/rspec/openapi/schema_builder.rb

```diff
@@ -71,6 +71,16 @@ def build_parameters(record)
     parameters
   end
 
+  def build_parameter_name(key, value)
+    key = key.to_s
+    if value.is_a?(Hash) && (value_keys = value.keys).size == 1
```

Comment:
1️⃣ build `{ "filter" => { "name" => "Benjamin" } }` as `filter[name]` instead of `filter`